packages/core/src/structures/IAgenticaVendor.ts

```diff
@@ -36,4 +37,18 @@ export interface IAgenticaVendor {
    * Options for the request.
    */
   options?: OpenAI.RequestOptions | undefined;
+
+  /**
+   * Number of concurrent requests allowed.
+   *
+   * If you configure this property, {@link Agentica} will constraint
+   * number of concurrent requests to the LLM vendor. If you want to
+   * share the semaphore instance with other agents, you can directly
+   * assign the {@link Semaphore} instance to this property.
+   *
+   * Otherwise, it will not limit the number of concurrent
+   * requests, and the {@link Agentica} will send requests
+   * asynchronously without any limit.
+   */
+  semaphore?: Semaphore | number | undefined;
```

Comment:
If it's something that requires user input, it's a good idea to specify it in the peerDependency as well.